### PR TITLE
Added two service 

### PR DESCRIPTION
**Pull request by InsaneNaman**
Function for custom emoji , user roles , Logout, login using localstorage

### CODE REVIEW

File: src/app/shared/REALtime API/auth/auth.service.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    );
+
+    const ping = this.subject$.find((data: (any)) => (data.msg === 'ping'));
+    const pong = this.subject$.find((data: (any)) => (data.msg === 'pong'));
+    const changed = this.subject$.find((data: (any)) => data.msg === 'changed');
+    const token = this.subject$.find((data: (any)) => data.msg === 'result');
+    this.subject$.next(JSON.stringify({'msg': 'connect', 'version': '1', 'support': ['1', 'pre2', 'pre1']}));
+
+
+    ping.subscribe(() => this.subject$.next('pong'));
```
Comment (ggazzo) [inline src/app/shared/REALtime API/auth/auth.service.ts]: the correct is send :  `this.subject$.next(JSON.stringify({msg:"pong"}))`

**Review by ggazzo — CHANGES_REQUESTED** [review]
Use an abstraction to reuse the send methods and avoid mistakes, you should generate a random id for each command sent and use the same connection for all services

File: src/app/shared/REALtime API/auth/auth.service.ts (hunk 1 above)
Comment (InsaneNaman) [reply src/app/shared/REALtime API/auth/auth.service.ts]: Okay 😄  Just give me some time, I'll do the changes

Comment (danielszenasi) [reply src/app/shared/REALtime API/auth/auth.service.ts]: https://github.com/RocketChat/Rocket.Chat.PWA/pull/23
Here is my suggestion for this problem

Comment (InsaneNaman) [reply src/app/shared/REALtime API/auth/auth.service.ts]: I know , daniel buddy you have opened a PR. I am working on the same or i can say we(me and @inf3cti0n95 ) also have found solution. Just give me some time , tomorrow or day after tomorrow we(you and me) will together merge our solution. Please, i just want some. 😄

**Comment by InsaneNaman** [comment]
- [ ] Connection(rxjs realtime package)

- [ ]  Request / Response Methods(rxjs realtime package)

- [ ] Subscriptions (rxjs realtime Package)

- [ ]  Apply UI system (Angular Material, Teradata Covalent(3rd party)) 

- [ ]  User login/registration 

- [ ] List channels

**Comment by InsaneNaman** [comment]
image
image

image

**Comment by InsaneNaman** [comment]
**Side Nav remaining**

image
image

**Pull request closed without merging**